synonymdev/bitkit-android
# POC wake2pay from notification

Resolves #synonymdev/bitkit/2038

- Handle data-only-notification for wake2pay POC
  Wakeup LDK node, sync it, connect peer and pay the `bolt11` invoice from the payload
  - All these OPs are done on the BG thread in a non-blocking fashion from within a `CoroutineWorker`, ie. one that supports suspending work.
- CLI app for testing FCM(Firebase Cloud Messages) in `./tools/fcm-tester/`
- `.editorconfig` for uniform formatting & cleanup of ldk code
---

> [!NOTE]
> Since the Android OS would kill a `Worker` that takes longer than allowed by the OS, the POC already demonstrates the most basic setup* for wake2pay shout run flawlessly.
> 
> `*` When the user has already had opened the native app and had a channel open with the LSP, all of which would be backed-up on the device by LDK/ldk-node and readily to be "reloaded" as it was done to test this PoC.

## To Test

- It is expected that you already had a channel with the default peer.
- Once that is in place, close the app, comment out the last 2 lines in `LauncherActivity.onCreate`, ie. `LauncherActivity.kt:21` and `LauncherActivity.kt:22`:
```kt
class LauncherActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initNotificationChannel()
        logFcmToken()
        // warmupNode(filesDir.absolutePath)
        // startActivity(Intent(this, MainActivity::class.java))
    }
}
```
- Run the app again and background it by going to the os' home screen.
- Check `./tools/fcm-tester/README.md` on how to send a remote message that should trigger the wake2pay.
- **Expect** `Result.Success` from the background worker.
